ehrapy/api/io/read.py

```diff
@@ -104,96 +105,111 @@ def _read(
+                raise NotImplementedError(f"There is currently no parser implemented for {extension} files!")
+            # cache results if wanted
+            if cache:
+                if not path_cache.parent.is_dir():
+                    path_cache.parent.mkdir(parents=True)
+                return DataReader._write_cache(raw_anndata, path_cache, columns_obs_only, index_column)  # type: ignore
+            return raw_anndata
 
     @staticmethod
     def _read_multiple_csv(  # noqa: N802
         filename: Path,
         delimiter: Optional[str] = None,
-        index_column: Union[Union[str, Optional[int]], Dict[str, Union[str, Optional[int]]]] = None,
+        index_column: Optional[Union[Dict[str, Union[str, int]], Union[str, int]]] = None,
         columns_obs_only: Union[Optional[List[Union[str]]], Dict[str, Optional[List[Union[str]]]]] = None,
         return_mudata_object: bool = False,
-    ) -> Union[MuData, Dict[str, AnnData]]:
-        """Read a dataset containing multiple files (in this case .csv or .tsv files).
+        cache: bool = False,
+    ) -> Tuple[Union[MuData, Dict[str, AnnData]], Dict[str, List[str]]]:
+        """Read a dataset containing multiple .csv/.tsv files.
 
         Args:
-            filename: File path to the directory containing multiple csvs dataset.
+            filename: File path to the directory containing multiple .csv/.tsv files.
             delimiter: Delimiter separating the data within the file.
-            index_column: Indices or column names of the index columns (obs)
-            columns_obs_only: List of columns per file (thus AnnData object) which should only be stored in .obs, but not in X. Useful for free text annotations.
-            return_mudata_object: If set to True, return a :class:`~mudata.MuData` object, else a dict of :class:`~anndata.AnnData` objects
+            index_column: Column names of the index columns for obs
+            columns_obs_only: List of columns per file (AnnData object) which should only be stored in .obs, but not in X. Useful for free text annotations.
+            return_mudata_object: When set to True, return a :class:`~mudata.MuData` object, otherwise a dict of :class:`~anndata.AnnData` objects
+            cache: Whether to cache results or not
 
         Returns:
-            An :class:`~mudata.MuData` object or a dict mapping the filename (object name) to the corresponding :class:`~anndata.AnnData` object.
+            An :class:`~mudata.MuData` object or a dict mapping the filename (object name) to the corresponding :class:`~anndata.AnnData` object and the columns
+            that are obs only for each object
         """
+        obs_only_all = {}
         if not return_mudata_object:
             anndata_dict = {}
         else:
             mudata = None
         for file in filename.iterdir():
-            if file.is_file() and file.suffix in [".csv", ".tsv"]:
+            if file.is_file() and file.suffix in {".csv", ".tsv"}:
```

Comment:
Why the change to curly brackets?

Reply:
Definitely not needed; i was just going over all lines , where we lookup anything in a collection and was refactoring it to set, if possible (for the faster lookup). But that's obviously not needed here, just to keep it consistent.

Reply:
I don't mind it :)